skyportal/handlers/api/annotation_services.py

```diff
@@ -19,6 +23,188 @@
+                  crossmatchLimmag:
+                    required: false
+                    type: number
+                    description: |
+                      Crossmatch limiting magnitude (for Gaia G mag).
+                      Will ignore sources fainter than this magnitude.
+                      If not specified, will use the default value in
+                      the config file, or None if not specified in the config.
+                      If zero or None, will take sources of any magnitude.
+                  group_ids:
+                    required: false
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups
+                      should be able to view annotation.
+                      Defaults to all of requesting user's groups.
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        obj = Obj.get_if_accessible_by(obj_id, self.current_user)
+        if obj is None:
+            return self.error('Invalid object id.')
+
+        data = self.get_json()
+
+        group_ids = data.pop('group_ids', None)
+        if not group_ids:
+            groups = self.current_user.accessible_groups
+        else:
+            try:
+                groups = Group.get_if_accessible_by(
+                    group_ids, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(
+                    'At least some of the groups are not accessible.', status=403
+                )
+
+        author = self.associated_user_object
+
+        catalog = data.pop('catalog', cfg['catalog'] or "gaiaedr3.gaia_source")
+        radius_arcsec = data.pop(
+            'crossmatchRadius', cfg['cross_match.gaia.radius'] or 2.0
+        )
+        limmag = data.pop('crossmatchLimmag', cfg['cross_match.gaia.limmag'])
+        num_matches = data.pop('crossmatchNumber', cfg['cross_match.gaia.number'] or 1)
+        candidate_coord = SkyCoord(ra=obj.ra * u.deg, dec=obj.dec * u.deg)
+
+        df = (
+            GaiaClass()
+            .cone_search(
+                candidate_coord,
+                radius_arcsec * u.arcsec,
+                table_name=catalog,
+            )
+            .get_data()
+            .to_pandas()
+        )
+
+        # first remove rows that have faint magnitudes
+        if limmag:  # do not remove if limmag is None or zero
+            df = df[df['phot_g_mean_mag'] < limmag]
+
+        # propagate the stars using Gaia proper motion
+        # then choose the closest match
+        if len(df) > 1:
+            df['adjusted_dist'] = 0  # new column
+            for index, row in df.iterrows():
+                c = SkyCoord(
```

Comment:
Nice

Reply:
It's nice because I took it from something Josh did for the offsets.